File: articles/cell-signaling-communication-biological-coordination/python/pathway_activation_summary.py
"""
Pathway activation summary.

Run:
    python python/pathway_activation_summary.py
"""

from __future__ import annotations

from pathlib import Path

import pandas as pd


ARTICLE_DIR = Path(__file__).resolve().parents[1]
PATHWAY_PATH = ARTICLE_DIR / "data" / "pathway_activation.csv"


def main() -> None:
    """Summarize pathway activation observations by context and pathway."""

    df = pd.read_csv(PATHWAY_PATH)

    context_summary = (
        df.groupby("cell_context")
        .agg(
            mean_activation=("activation_score", "mean"),
            max_activation=("activation_score", "max"),
            n_observations=("activation_score", "size"),
        )
        .reset_index()
        .sort_values("mean_activation", ascending=False)
    )

    pathway_summary = df.sort_values("activation_score", ascending=False)

    print(context_summary.round(4).to_string(index=False))
    print(pathway_summary.round(4).to_string(index=False))


if __name__ == "__main__":
    main()

File: articles/cell-signaling-communication-biological-coordination/r/pathway_activation_summary.R
# Pathway activation summary workflow in R.

library(dplyr)

pathway_path <- file.path("data", "pathway_activation.csv")

if (!file.exists(pathway_path)) {
  pathway_path <- file.path("..", "data", "pathway_activation.csv")
}

pathway_df <- read.csv(pathway_path)

context_summary <- pathway_df %>%
  group_by(cell_context) %>%
  summarise(
    mean_activation = mean(activation_score),
    max_activation = max(activation_score),
    n_observations = n(),
    .groups = "drop"
  ) %>%
  arrange(desc(mean_activation))

print(context_summary)
print(pathway_df %>% arrange(desc(activation_score)))
